docs/operations.rst

```diff
@@ -14,42 +14,79 @@ Provisioning
-`.pgpass` file.
+	During the restart of the monitor, the other nodes might have trouble
+	connecting to the monitor. The ``pg_autoctl`` command is designed to
+	retry connecting to the monitor and handle errors gracefully.
 
-  1. health check connection from monitor for `autoctl` user to both `postgres` and `pg_auto_failover` databases.
-  2. connections for `pg_autoctl` command from data nodes to monitor for `autoctl_node` user.
-  3. replication connections from secondary to primary data nodes for `replication` user.
-     Notice that primary and secondary nodes change during failover. Thus this setting
-     should be done on both primary and secondary nodes.
-  4. settings need to be updated after a new node is added.
+  - on the Postgres nodes, the ``pg_autoctl`` command connects to the
+    monitor every once in a while (every second by default), and then calls
+    the ``node_active`` protocol, a stored procedure in the monitor databases.
 
-See `PostgreSQL documentation`__ on setting up `.pgpass` file.
+	The ``pg_autoctl`` also verifies at each connection to the monitor that
+	it's running the expected version of the extension. When that's not the
+	case, the "node-active" sub-process quits, to be restarted with the
+	possibly new version of the ``pg_autoctl`` binary found on-disk.
 
-__ https://www.postgresql.org/docs/current/libpq-pgpass.html
+As a result, here is the standard upgrade plan for pg_auto_failover:
 
+  1. Upgrade the pg_auto_failover package on the all the nodes, monitor
+     included.
 
-Operations
-----------
+	 When using a debian based OS, this looks like the following command for
+	 pg_auto_failover 1.3.1::
+
+	   sudo apt-get remove pg-auto-failover-cli-enterprise-1.0 postgresql-11-auto-failover-enterprise-1.0
+	   sudo apt-get install -q -y pg-auto-failover-cli-enterprise-1.3 postgresql-11-auto-failover-enterprise-1.3
+
+  2. Restart the ``pgautofailover`` service on the monitor.
+
+	 When using the systemd integration, all we need to do is::
+
+	   sudo systemctl restart pgautofailover
+
+	 Then we may use the following commands to make sure that the service is
+	 running as expected::
+
+	   sudo systemctl status pgautofailover
+	   sudo journalctl -u pgautofailover
+
+	 At this point it is expected that the ``pg_autoctl`` logs show that an
+	 upgrade has been performed by using the ``ALTER EXTENSION
+	 pgautofailover UPDATE TO ...`` command. The monitor is ready with the
+	 new version of pg_auto_failover.
+
+When the Postgres nodes ``pg_autoctl`` process connects to the new monitor
+version, the check for version compatibility fails, and the "node-active"
+sub-process exits. The main ``pg_autoctl`` process supervisor then restart
+the "node-active" sub-process from its on-disk binary executable file, which
+has been upgraded to the new version. That's why we first install the new
+packages for pg_auto_failover on every node, and only then restart the
+monitor.
```

Comment:
Probably good to note here (and/or at the top of this step) that not updating pg_autoctl on Postgres nodes will result in postgres going down.

Reply:
Added a note about that, thanks!